library/std/src/os/linux/net.rs

```diff
@@ -0,0 +1,70 @@
+//! Linux-specific extensions to primitives in the [`std::net`] module.
+//!
+//! [`std::net`]: crate::net
+
+use crate::io;
+use crate::net;
+use crate::sealed::Sealed;
+use crate::sys_common::AsInner;
+
+/// Os-specific extensions for [`TcpStream`]
+///
+/// [`TcpStream`]: net::TcpStream
+#[unstable(feature = "tcp_quickack", issue = "96256")]
+pub trait TcpStreamExt: Sealed {
+    /// Sets the value of the `TCP_QUICKACK` option on this socket.
+    ///
+    /// If set, acks are sent immediately, rather than delayed if needed in
+    /// accordance to normal TCP operation. This flag is not permanent, it
+    /// only enables a switch to or from quickack mode.  Subsequent operation
+    /// of the TCP protocol will once again enter/leave quickack mode
+    /// depending on internal protocol processing and other factors.
+    /// (since Linux 2.4.4)
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// #![feature(tcp_quickack)]
+    /// use std::net::TcpStream;
+    /// use std::os::linux::net::TcpStreamExt;
+    ///
+    /// let stream = TcpStream::connect("127.0.0.1:8080")
+    ///         .expect("Couldn't connect to the server...");
+    /// stream.set_quickack(true).expect("set_quickack call failed");
+    /// ```
+    #[unstable(feature = "tcp_quickack", issue = "96256")]
+    fn set_quickack(&self, quickack: bool) -> io::Result<()>;
+
+    /// Gets the value of the `TCP_QUICKACK` option on this socket.
+    ///
+    /// For more information about this option, see [`TcpStreamExt::set_quickack`].
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// #![feature(tcp_quickack)]
+    /// use std::net::TcpStream;
+    /// use std::os::linux::net::TcpStreamExt;
+    ///
+    /// let stream = TcpStream::connect("127.0.0.1:8080")
+    ///         .expect("Couldn't connect to the server...");
+    /// stream.set_quickack(true).expect("set_quickack call failed");
+    /// assert_eq!(stream.quickack().unwrap_or(false), true);
+    /// ```
```

Comment:
Since these tests are `no_run` it would be good to have actual tests in a test module.

Reply:
can I also just remove the `no_run`?

Reply:
I don't know how rustdoc tests interact with cfgs and PR CI only runs linux tests, so it were to fail it would likely only fail after a merge.

But if you have some non-linux machine you could try and see what happens there?

Reply:
ah yes it failed already, I found one test for the `nodelay` and adapted it for `quickack`. Hope this works